### PR TITLE
[DONOTMERGE] Update API smoke tests

### PR DESCRIPTION
**Pull request by elyezer**
Update API smoke tests to use the latest available tools from NailGun.
Now we can work on a higher level of abstraction.

```
$ nosetests --nologcapture tests/foreman/smoke/test_api_smoke.py
........
----------------------------------------------------------------------
Ran 8 tests in 443.280s

OK
```

This PR depends on ~~#2576~~ and new NailGun version.

### CODE REVIEW

**Comment by Ichimonji10** [comment]
Niiiice. :+1:

**Comment by sthirugn** [comment]
code looks neat now.  ACK

**Comment by sthirugn** [comment]
Lets get this in once nailgun arrives so I can trigger tests again.

File: tests/foreman/smoke/test_api_smoke.py [hunk 1]
```diff
@@ -1009,11 +1012,11 @@ def test_end_to_end(self):
...
-        product = "Red Hat Enterprise Linux Server"
-        reposet = ("Red Hat Enterprise Virtualization Agents "
-                   "for RHEL 6 Server (RPMs)")
-        repo = ("Red Hat Enterprise Virtualization Agents for RHEL 6 Server "
-                "RPMs x86_64 6Server")
+        product = 'Red Hat Enterprise Linux Server'
+        reposet = ('Red Hat Enterprise Virtualization Agents '
+                   'for RHEL 6 Server (RPMs)')
+        repo = ('Red Hat Enterprise Virtualization Agents for RHEL 6 Server '
+                'RPMs x86_64 6Server')
```
Comment (abalakh) [inline tests/foreman/smoke/test_api_smoke.py]: Bonus karma for using common constants ```PRDS['rhel']```, ```REPOSET['rhva6']``` and adding repo to ```REPOS``` dict :)

Comment (elyezer) [inline tests/foreman/smoke/test_api_smoke.py]: Thanks for pointing that out. I will definitely use them!

**Comment by abalakh** [comment]
ACK

File: tests/foreman/smoke/test_api_smoke.py [hunk 2]
```diff
@@ -1084,20 +1080,10 @@ def test_end_to_end(self):
...
-            )
-            self.assertEqual(result.return_code, 0)
-            # Register client with foreman server using activation-key
-            result = vm.run(
-                u'subscription-manager register --activationkey {0} '
-                '--org {1} --force'
-                .format(activation_key_name, org.label)
-            )
+            vm.install_katello_cert()
+            result = vm.register_contenthost(activation_key_name, org.label)
```
Comment (Ichimonji10) [inline tests/foreman/smoke/test_api_smoke.py]: :+1:

**Comment by Ichimonji10** [comment]
NailGun 0.18.2 has been released and is being used by Robottelo (#2583). #2576 has been merged. This PR is unblocked.

**Comment by Ichimonji10** [comment]
ACK

**Pull request merged**